compiler/rustc_mir_build/src/check_unsafety.rs

```diff
@@ -254,7 +254,9 @@ impl<'a, 'tcx> Visitor<'a, 'tcx> for UnsafetyVisitor<'a, 'tcx> {
                         );
                     };
                     match borrow_kind {
-                        BorrowKind::Shallow | BorrowKind::Shared | BorrowKind::Unique => {
```

Comment:
I think `BorrowKind::Unique` actually belongs to `BorrowKind::Mut` ?

Reply:
yes, it should always require `unsafe`, the status quo is wrong